framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -543,6 +543,17 @@ public abstract class SourceChecker extends AbstractTypeProcessor implements Opt
   /** List of upstream checker names. Includes the current checker. */
   protected @MonotonicNonNull List<@FullyQualifiedName String> upstreamCheckerNames;
 
+  /** True if the -Afilenames command-line argument was passed. */
+  private boolean printFilenames;
+  /** True if the -Awarns command-line argument was passed. */
+  private boolean warns;
+  /** True if the -AshowSuppressWarningsStrings command-line argument was passed. */
+  private boolean showSuppressWarningsStrings;
+  /** True if the -ArequirePrefixInWarningSuppressions command-line argument was passed. */
+  private boolean requirePrefixInWarningSuppressions;
+  /** True if the -AwarnUnneededSuppressions command-line argument was passed. */
+  private boolean warnUnneededSuppressions;
+
   @Override
   public final synchronized void init(ProcessingEnvironment env) {
```

Comment:
After a lot of digging, I figured out that `init` is not called on subcheckers. So all the logic in this method that should happen for all checkers needs to be moved to `initChecker`.